### PR TITLE
fixed memory leak due to mesh not being cleared

### LINKED ISSUE #482: Memory Leak using the plotter
Hello I made a small app in notebook using pyvista and panel to plot some temporal data.
(A FEM code save a .vtk of results at  certain times in a directory)
I use pyvista to load the .vtk wich are unstructured grids and I add it to a pyvista plotter to create a renderWindow which I use in panel to plot the results
My pyvista version is : '0.22.4'
Basically the result looks like this a notebook:
ezgif com-video-to-gif (22)

The code to achieve this is more or less (for the real case I made some optimisations):
```python
import panel as pn
import pyvista as pv
from functools import partial

pl = pv.Plotter()
vtkpan = pn.pane.VTK(sizing_mode='stretch_both', orientation_widget=True)
cbpan = pn.pane.Bokeh(sizing_mode='stretch_both')
player = pn.widgets.Player(name='Discrete Player', start=0, end=72, value=0, loop_policy='loop', interval=1000)

def update_visu3d(vtkpan, cbpan, pl, nb_file):
    pl.clear()
    pl.add_mesh(pv.read('vtk_creation/Results/VTK/MacroElem0_{:d}.vtk'.format(nb_file)))
    if vtkpan.camera:
        pl.camera_position = [vtkpan.camera[k] for k in ['position', 'focalPoint', 'viewUp']]
    vtkpan.object = pl.ren_win
    cbar = vtkpan.construct_colorbars()
    cbar.below[0].padding=20
    cbpan.object = cbar
    
update_fun = partial(update_visu3d, vtkpan=vtkpan, cbpan=cbpan, pl=pl)

def callback(*events):
    update_fun(nb_file=events[0].new)

player.param.watch(callback, ['value'], onlychanged=True)
col = pn.Column(pn.Row(player, cbpan), vtkpan, sizing_mode='stretch_both')
update_fun(nb_file=player.value)
col
```

However I suffer from memory leak:
This memory leak can be easily reproduce with a dummy example:
```python
import pyvista as pv
import numpy as np
pl=pv.Plotter(off_screen=True)

def create_mesh():
    values = np.random.rand(1000).reshape((20, 5, 10))
    values.shape
    grid = pv.UniformGrid()
    grid.dimensions = np.array(values.shape) + 1
    # Edit the spatial reference
    grid.origin = (100, 33, 55.6)  # The bottom left corner of the data set
    grid.spacing = (1, 5, 2)  # These are the cell sizes along each axis
    # Add the data values to the cell data
    grid.cell_arrays["values"] = values.flatten(order="F")  # Flatten the array!
    return grid

for nb_file in range(10000):
    pl.clear()
    grid = create_mesh()
    pl.add_mesh(grid)
```
ezgif com-video-to-gif (23)

I suppose even if actor is removed from the plotter when `pl.clear` is called, some references to this actor or actor elements are kept somehwere

### PR DESCRIPTION
**Pull request by akaszynski**
Super easy - fixes #482.

### CODE REVIEW

**Comment by akaszynski** [comment]
It pains me how long travis CI takes compared to Azure.

**Comment by banesullivan** [comment]
The long builds on Travis are due to the docs; running all the examples in the gallery takes a while... 

We could stop building them unless on the `master` branch, but a lot of code is covered by those examples so that could get risky

File: pyvista/plotting/renderer.py [hunk 1]
```diff
@@ -585,6 +585,17 @@ def camera_position(self):
                 self.camera.GetFocalPoint(),
                 self.camera.GetViewUp()]
 
+    def clear(self):
+        """removes all actors and properties"""
```
Comment (banesullivan) [inline pyvista/plotting/renderer.py]:
```suggestion
        """Remove all actors and properties."""
```

**Comment by akaszynski** [comment]
> The long builds on Travis are due to the docs; running all the examples in the gallery takes a while...

> We could stop building them unless on the master branch, but a lot of code is covered by those examples so that could get risky

We might consider separating out the docstring testing, but otherwise I think we should continue the tests as it might reveal some issue not covered in the unit testing.  I see the examples as both functional testing and documentation.

**Comment by akaszynski** [comment]
@banesullivan, thanks for that, was about to fix it.  Forgot about `pydocstyle`.  It's good we have that in there, and I might start adding that to my other open and closed source projects.

**Comment by banesullivan** [comment]
> I see the examples as both functional testing and documentation.

Same here - they are a good set of more complex functional tests than what's covered in the `tests` module. Unfortunately, this makes the long build times worthwhile.

**Comment by banesullivan** [comment]
Failed build is unrelated to these changes

**Pull request merged**